### ISSUE TITLE
[gatsby-source-wordpress] permalinks should not be set to Plain. 

### ISSUE THREAD

**Issue by dlaub3** [gatsbyjs/gatsby#1871]
This is as much an FYI for those having trouble as anything. I found through trial and error that the gatsby-source-wordpress plugin requires the permalinks to be set to something other than 'Plain' which has the url formatted like this 'www.yoursite.com?p=123'.

**Comment by sebastienfi** [comment]
Hi @dlaub3 
The config on the example site is the default for WordPress which also happens to be the most widely used.

Could you please report the problem you've encountered with defining the WP Permalink's setting to "Plain" ?

Current default config : 
![image](https://user-images.githubusercontent.com/5808108/29712690-584c300e-899b-11e7-8765-ecc622ac804c.png)

**Comment by dlaub3** [comment]
Hey, I setup a local WP instance with docker connecting to it through
localhost on port 80. When I installed it, it the permalinks were set to
"Plain" which is the very top option in the image you provided. I ran
Gastby against it but it didn't work, so I tried a few other blogs and then
a different local WP running docker with the same setup. Running Gatsby
against my other local dockerized WP instance worked fine. So I looked into
the different settings and discovered that the permalinks were different.
Setting the permalinks to "Post Name" worked, but if I used "Plain" Gatsby
wouldn't import the posts.

**Comment by sebastienfi** [comment]
Was it any error message displayed ?

**Comment by dlaub3** [comment]
Yes, I'll run the setup again and see what it said exactly. I believe
something was null or undefined. I'm assuming because no data was pulled
from WP. I'll let you know for sure.


On Fri, Aug 25, 2017 at 11:51 AM, Sébastien Fichot <notifications@github.com

**Comment by sebastienfi** [comment]
I ran a test with the setting you mentionned but got no error.
Would be great to reproduce. Maybe fixed by latests versions ?

**Comment by dlaub3** [comment]
This is the output from `npm run develop`. 

```success delete html files from previous builds — 0.011 s
success open and validate gatsby-config.js — 0.004 s
success copy gatsby files — 0.019 s
error Plugin gatsby-source-wordpress returned an error


  TypeError: Cannot convert undefined or null to object
  
  - Function.keys
  
  - gatsby-node.js:610 getValidRoutes
    [gatsby-eas]/[gatsby-source-wordpress]/gatsby-node.js:610:32
  
  - gatsby-node.js:460 _callee$
    [gatsby-eas]/[gatsby-source-wordpress]/gatsby-node.js:460:27
  
  - next_tick.js:131 _combinedTickCallback
    internal/process/next_tick.js:131:7
  
  - next_tick.js:180 process._tickCallback
    internal/process/next_tick.js:180:9
  

success source and transform nodes — 0.832 s
success building schema — 0.125 s
success createLayouts — 0.022 s
success createPages — 0.005 s
success createPagesStatefully — 0.006 s
GraphQL Error Unknown field `wordpressPage` on type `RootQueryType`

  file: /home/nokan/Documents/projects/gatsby-eas/src/pages/index.js

   1 | 
   2 |   query homePageQuery {
>  3 |     wordpressPage(id: { eq: "PAGE_1" }) {
     |     ^
   4 |       id
   5 |       title
   6 |       content
   7 |       excerpt
   8 |       date
   9 |       date_gmt
  10 |       modified
  11 |       modified_gmt
  12 |       slug
  13 |       status

success extract queries from components — 0.119 s
success run graphql queries — 0.003 s
success write out page data — 0.003 s
success update schema — 0.067 s

info bootstrap finished - 2.964 s

 DONE  Compiled successfully in 17990ms
```

Steps to reproduce: 
Install Gatsby starter default 
Add WordPress plugin
Change  baseUrl:  in gatsby-config.js to local WordPress install. 
If permalinks are set to 'plain' the above error occurs. 



Here is my package.json 

```
{
  "name": "gatsby-starter-default",
  "description": "Gatsby default starter",
  "version": "1.0.0",
  "author": "Kyle Mathews <mathews.kyle@gmail.com>",
  "dependencies": {
    "gatsby": "^1.8.12",
    "gatsby-link": "^1.6.8",
    "gatsby-plugin-react-helmet": "^1.0.3",
    "gatsby-source-wordpress": "^1.7.6",
    "hedron": "^0.7.1",
    "striptags": "^3.0.1",
    "styled-components": "^2.1.2"
  },
  "keywords": [
    "gatsby"
  ],
  "license": "MIT",
  "main": "n/a",
  "scripts": {
    "build": "gatsby build",
    "develop": "gatsby develop",
    "format": "prettier --trailing-comma es5 --no-semi --single-quote --write 'src/**/*.js'",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "devDependencies": {
    "prettier": "^1.5.2"
  }
}
```

Let me know if I can provide any other information for you.

**Comment by dlaub3** [comment]
Interestingly I noticed today that the WordPress rest api doesn't work when set to 'plain permalinks'. This must be the cause of my problem.

**Comment by sebastienfi** [comment]
@dlaub3 That's weird. What do you see when you call `your-url/wp-json/wp/v2/pages` ?

**Comment by KyleAMathews** [comment]
What is the status here? Is there something we can do still here?

**Comment by KyleAMathews** [comment]
Looks like this was fixed in #5053 so closing.
